app/models/king.rb

```diff
@@ -1,2 +1,11 @@
 class King < Piece
+	def valid_move?(new_rank, new_file)
+		rank_diff = (new_rank - rank).abs
+		file_diff = (new_file - file).abs
+		if (rank_diff <= 1) && (file_diff <= 1) && !is_move_obstructed?(new_rank, new_file) && !is_move_malformed?(new_rank, new_file)
+			return true			
+		else
+			return false
```

Comment:
Optional: Ruby methods always return the value of the last statement in the method. In this case, the last line will be either `true` or `false` depending on the outcome of the conditional, which is what you want to return anyway, so it's not necessary to use an explicit `return`.